icu4c/source/common/uniset_props.cpp

```diff
@@ -1107,7 +1131,7 @@ void UnicodeSet::parseElements(Lexer &lexer,
                                                      lexer.lookahead2().debugString(),
                                                  lexer, ec);
         }
-    } else if (lexer.lookahead().codePoint().has_value()) {
+    } else if (lexer.lookahead().codePoint().has_value() && !lexer.lookahead().isBracketedElement()) {
```

Comment:
Note: #3604 inadvertently caused `[a-{z}]` (but not `[{a}-z]`!) to become well-formed. We want it to become well-formed eventually, but this needs a proposal. This drive-by change restores the ICU 78 behaviour (and I am adding a test that would have caught this).